python/tank/api.py

```diff
@@ -274,9 +274,15 @@ def template_from_path(self, path):
             return matched[0]
         else:
             # ambiguity!
+            # We're erroring out anyway, take the time to create helpful debug info!
+            fields = []
+            for template in matched:
+                fields.append(template.validate_and_get_fields(path))
+
             msg = "%d templates are matching the path '%s'.\n" % (len(matched), path)
             msg += "The overlapping templates are:\n"
-            msg += "\n".join([str(x) for x in matched])
+            for f, t in zip(fields, matched):
```

Comment:
is t and f for field and template? The code is easier to read if these variables are more descriptive.